### DISCUSSION TITLE
Can you keep multiple models loaded in CPU RAMs (not VRAM)?

### DISCUSSION THREAD

**Discussion by Sunsphire** [vladmandic/sdnext#2639] [Q&A] [answered]
Might be a stupid question but I'm curious if it's possible. Exact example of my situation:

I use a model/checkpoint to generate my image and I use a separate model/checkpoint for ADetailer - because the initial model can't do caucasian faces for example. This work great but when the second model is changed when ADetailer do it's work it takes a significant amount of time to change between models. I also use 4 or 5 ADetailer tabs/models (face_yolov8n, mediapipe_face_full, eyes, hands, etc) so the amount of time adds up. I see the memory going up every time it change the model/checkpoint but going down after that. I also noticed that it's happening every time ADetailer go to the next step...so it's happening once for face_yolo and then again for mediapipe, eyes, etc, even if all ADetailers use the same second checkpoint.

Is it possible to keep 2 models loaded in ram? In my mind that would help everything load faster...and I have plenty of ram but not VRAM sadly.

Please let me know if my explanation is not clear, I'll try to do a better job

**Comment by vladmandic** [comment, marked as answer]
in settings:
![image](https://github.com/vladmandic/automatic/assets/57876960/c41dfa93-20f6-456d-9584-863ac8f9ef7c)
